sonar-python-plugin/src/main/java/org/sonar/plugins/python/indexer/SonarQubePythonIndexer.java

```diff
@@ -183,4 +187,18 @@ public boolean canBeScannedWithoutParsing(InputFile inputFile) {
   public CacheContext cacheContext() {
     return caching.cacheContext();
   }
+
+  private static String getCacheVersion(SensorContext context) {
+    String implementationVersion = getImplementationVersion(SonarQubePythonIndexer.class);
+    return context.config().get(PYTHON_VERSION_KEY).map(v -> implementationVersion + ";" + v).orElse(implementationVersion);
+  }
+
+  private static String getImplementationVersion(Class<?> cls) {
+    String implementationVersion = cls.getPackage().getImplementationVersion();
+    if (implementationVersion == null) {
+      LOG.warn("Implementation version of the Python plugin not found. Cached data may not be invalidated properly.");
```

Comment:
Maybe we could add "Cached data may not be invalidated properly , hence analysis results may be inaccurate."